src/main/java/com/amazonaws/services/kinesis/clientlibrary/lib/worker/ShardConsumer.java

```diff
@@ -51,6 +51,7 @@
     private final KinesisDataFetcher dataFetcher;
     private final IMetricsFactory metricsFactory;
     private final ILeaseManager<KinesisClientLease> leaseManager;
+    private final KinesisClientLibLeaseCoordinator leaseCoordinator;
```

Comment:
~KinesisClientLibLeaseCoordinator~ `LeaseCoordinator<KinesisClientLease>` ?

Reply:
This is to match the field in Worker. Is there a particular reason we should use LeaseCoordination<KinesisClientLease> ?

Reply:
if LeaseCoordinator is not exposing leaseManager, then we can stick with KinesisClientLibLeaseCoordinator